6 audio lattice

messiaen opened the pull request:
fixes #6

messiaen committed:
Start LatticeFieldMapper

messiaen committed:
Style cleanup

messiaen committed:
Clean up LatticeFormat selection

messiaen committed:
Set slop based on position_increment_seconds

    - the math here isn't quite right as of this commit.  It should
      reflect the below.
    - setting 'slop_secs' to 1.5, means that the sum of the time gaps
      between all the tokens in a matching span can be at most 1.5 secs.
      (e.g. if we have tokens start times at 0.0, 0.5, 1.5, 2.0; the
      first three token could be in a matching span but the forth could
      not.

messiaen committed:
Fix position increment and slop calculations

messiaen committed:
Fix position increment

    - First token is special

messiaen committed:
Test LatticeTokenFilter

messiaen committed:
Test LatticeQueryBuilder

messiaen committed:
Test query with yaml itests

messiaen committed:
Test base lattice and scores

messiaen committed:
Class renames and change default payload fun to sum

messiaen committed:
Test LatticeTokenFilter payloads

messiaen committed:
Norm match_lattice scores by length * lenNormFactor

mjwall commented on Makefile:
```diff
@@ -4,10 +4,11 @@ MAKEFLAGS += --no-builtin-variables
 
 ES_VERSION = $(shell grep ES_VERSION .env | cut -d '=' -f 2)
 VERSION = $(shell grep PLUGIN_VERSION .env | cut -d '=' -f 2)
+COMMIT_HASH = $(shell grep COMMIT_HASH .env | cut -d '=' -f 2)
```
Maybe something like 
```
git describe --match=NeVeRmAtCh --always --abbrev=8 --dirty
```

messiaen replied:
fixed

mjwall commented on src/main/java/com/eigendomain/eslatticeindex/index/LatticeWhitespaceTokenizer.java:
```diff
@@ -0,0 +1,70 @@
+ *
+ * @see UnicodeWhitespaceTokenizer
+ */
+public final class LatticeWhitespaceTokenizer extends CharTokenizer {
```
I asked why this was needed, you said to work around the WhitespaceTokenizer not being loaded in tests.  Just making a note

messiaen replied:
I've removed the custom class in favor of calling using the normal WhitespaceTokenizer in the custom factory.  This is still not great, but it a little better at least. Created #15 to fix this.

mjwall commented on Makefile:
```diff
@@ -26,6 +27,7 @@ push:
 test:
 	./gradlew cleanTest test
```
Not running `./gradlew integTest` anywhere.   Consider running tests before the assemble

mjwall commented on Makefile:
```diff
@@ -4,10 +4,11 @@ MAKEFLAGS += --no-builtin-variables
 
 ES_VERSION = $(shell grep ES_VERSION .env | cut -d '=' -f 2)
 VERSION = $(shell grep PLUGIN_VERSION .env | cut -d '=' -f 2)
```
Maybe should bump the version in .env

messiaen committed:
Rm custom whitespace filter and add comments

messiaen committed:
Makefile fixes

    - actually record git hash
    - run itests in test target
    - remove build_plugin target

messiaen committed:
Bump version to 2.0.0-rc2

messiaen committed:
Fix docker image name in docker-compose
